src/states/proving_node.rs

```diff
@@ -0,0 +1,601 @@
+            timer,
+            resource_prover,
+            disable_resource_proof: dev_config.disable_resource_proof,
+            joining_prefix: our_section.0,
+            notified_nodes: Default::default(),
+        };
+        node.start(our_section.1, &proxy_pub_id);
+        node
+    }
+
+    /// Called immediately after construction. Sends `ConnectionInfoRequest`s to all members of
+    /// `our_section` to then start the candidate approval process.
+    fn start(&mut self, our_section: BTreeSet<PublicId>, proxy_pub_id: &PublicId) {
+        self.resource_prover.start(self.disable_resource_proof);
+
+        trace!("{} Relocation completed.", self);
+        info!(
+            "{} Received relocation section. Establishing connections to {} peers.",
+            self,
+            our_section.len()
+        );
+
+        let src = Authority::Client {
+            client_id: *self.full_id.public_id(),
+            proxy_node_name: *proxy_pub_id.name(),
+        };
+        // There will be no events raised as a result of these calls, so safe to just use a
+        // throwaway `EventBox` here.
+        let mut outbox = EventBuf::new();
+        for pub_id in &our_section {
+            debug!(
+                "{} Sending connection info request to {:?} on Relocation response.",
+                self, pub_id
+            );
+            let dst = Authority::ManagedNode(*pub_id.name());
+            if let Err(error) = self.send_connection_info_request(*pub_id, src, dst, &mut outbox) {
+                debug!(
+                    "{} - Failed to send connection info request to {:?}: {:?}",
+                    self, pub_id, error
+                );
+            }
+        }
+    }
+
+    pub fn handle_action(&mut self, action: Action, outbox: &mut EventBox) -> Transition {
+        match action {
+            Action::ClientSendRequest { result_tx, .. }
+            | Action::NodeSendMessage { result_tx, .. } => {
+                // TODO (adam): should we backlog `NodeSendMessage`s and handle them once we are
+                //              approved?
+                let _ = result_tx.send(Err(InterfaceError::InvalidState));
+            }
+            Action::Id { result_tx } => {
+                let _ = result_tx.send(*self.id());
+            }
+            Action::Timeout(token) => {
+                if let Transition::Terminate = self.handle_timeout(token, outbox) {
+                    return Transition::Terminate;
+                }
+            }
+            Action::ResourceProofResult(pub_id, messages) => {
+                let msg = self
+                    .resource_prover
+                    .handle_action_res_proof(pub_id, messages);
+                self.send_direct_message(pub_id, msg);
+            }
+            Action::Terminate => {
+                return Transition::Terminate;
+            }
+        }
+
+        Transition::Stay
+    }
+
+    pub fn handle_crust_event(
```

Comment:
Same for this, a lot of copy paste. Not sure we can do anything about it.

Reply:
Same as above: https://github.com/maidsafe/routing/pull/1613#discussion_r276741015

Reply:
Yes, it was more something that showed up in this diff that is likely to cause issue in the future, but we probably cannot really do anything at this point.